### PR TITLE
Update cluster-api dependency

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

This PR updates the cluster-api upstream dependency and the rest of the vendor folder, it also removes glog and fixes some breaking changes. The machine objects now reference the cluster object by using `cluster.k8s.io/cluster-name` label.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by vincepri** [comment]
Waiting for https://github.com/kubernetes/test-infra/pull/11290

File: cmd/clusterctl/main.go [hunk 1]
```diff
@@ -17,27 +17,13 @@ limitations under the License.
...
-	flags := flag.NewFlagSet("klog", flag.ExitOnError)
-	klog.InitFlags(flags)
-	flags.Set("alsologtostderr", "true")
-	flags.Set("v", "4")
-	flag.Parse()
-}
-
 func main() {
-	initLogs()
 	var err error
```
Comment (justinsb) [inline cmd/clusterctl/main.go]: We don't need klog.InitFlags(nil) here?

File: pkg/cloud/google/machineactuator.go [hunk 2]
```diff
@@ -28,21 +28,19 @@ import (
 	"strings"
 	"time"
 
-	"github.com/golang/glog"
+	"github.com/ghodss/yaml"
```
Comment (justinsb) [inline pkg/cloud/google/machineactuator.go]: FYI I think we have `sigs.k8s.io/yaml` that is our preferred fork...

Comment (vincepri) [reply pkg/cloud/google/machineactuator.go]: +1

File: cmd/clusterctl/main.go (hunk 1 above)
Comment (vincepri) [reply cmd/clusterctl/main.go]: The flags are already initialized upstream as part of `cmd.Execute`.

**Comment by vincepri** [comment]
@justinsb Rebased, updated with feedback and pushed :)

**Comment by vincepri** [comment]
@justinsb @roberthbailey ptal.

File: Dockerfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ COPY cmd/    cmd/
 COPY vendor/ vendor/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager sigs.k8s.io/cluster-api-provider-gcp/cmd/manager
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -ldflags '-extldflags "-static"' -o manager sigs.k8s.io/cluster-api-provider-gcp/cmd/manager
```
Comment (justinsb) [inline Dockerfile]: It is weird that we do this, but ... I agree.  The `cc` distroless image does include libc and thus avoids the need for static linking, but this feels better.  I think there's some limitations around static linking on osx, but we're in docker anyway...

**Comment by justinsb** [comment]
Thanks @vincepri 

/approve
/lgtm

**Pull request merged**